### PR TITLE
fix: rewrite powershell scripts to use PSScriptRoot

### LINKED ISSUE #7417: [BUG] npm ci fails on Windows with Node.js 22.0.0
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

On Windows, `npm ci` fails.

(Note: this works fine on Linux and macOS)

https://github.com/westy92/holiday-event-api-js/actions/runs/8827184894/job/24234452752

```
Run npm ci
  npm ci
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
node:internal/modules/cjs/loader:1[2](https://github.com/westy92/holiday-event-api-js/actions/runs/8827184894/job/24234452752#step:4:2)05
  throw err;
  ^

Error: Cannot find module 'C:\npm\prefix\node_modules\npm\bin\npm-cli.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1202:15)
    at Module._load (node:internal/modules/cjs/loader:1027:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:187:1[4](https://github.com/westy92/holiday-event-api-js/actions/runs/8827184894/job/24234452752#step:4:5))
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v22.0.0
```

### Expected Behavior

`npm ci` succeeds

### Steps To Reproduce

1. on GitHub hosted runners
2. running windows
3. running Node 22.0.0
4. run `npm ci`

### Environment

- npm: 10
- Node.js: 22.0.0
- OS Name: Windows
- System Model Name: GitHub Windows Runner
- npm config:
```ini
; copy and paste output from `npm config ls` here
```

### PR DESCRIPTION
**Pull request by lukekarrys**
Fixes #7417
Ref https://github.com/nodejs/node/issues/52682

### CODE REVIEW

File: bin/npm.ps1 [hunk 1]
```diff
@@ -1,35 +1,25 @@
...
-  $input | & $nodeexe $npmprefixclijs $args
+  $input | & $NODE_EXE $NPM_CLI_JS $args
 } else {
-  & $nodeexe $npmprefixclijs $args
+  & $NODE_EXE $NPM_CLI_JS $args
 }
-$ret=$LASTEXITCODE
-exit $ret
+
+$RET=$LASTEXITCODE
```
Comment (wraithgar) [inline bin/npm.ps1]: This may be a chesterton's fence but why does `$RET` exist?

**Review by wraithgar — APPROVED** [review]

File: bin/npm.ps1 (hunk 1 above)
Comment (lukekarrys) [reply bin/npm.ps1]: Removed this in favor of just `exit $LASTEXITCODE`

File: bin/npm.ps1 [hunk 2]
```diff
@@ -1,35 +1,28 @@
...
-$basedir=Split-Path $MyInvocation.MyCommand.Definition -Parent
 
-$exe=""
-if ($PSVersionTable.PSVersion -lt "6.0" -or $IsWindows) {
-  # Fix case when both the Windows and Linux builds of Node
-  # are installed in the same directory
-  $exe=".exe"
+$NODE_EXE="$PSScriptRoot/node.exe"
+if (-not (Test-Path $NODE_EXE)) {
+  $NODE_EXE="$PSScriptRoot/node"
```
Comment (wraithgar) [inline bin/npm.ps1:5]: `npm.cmd` doesn't do this. It just checks for `node.exe` in `%~dp0` and then falls back to `node` with no path.  Is there a reason we do this interim step?

Comment (lukekarrys) [reply bin/npm.ps1:5]: The previous version of `bin/npm.ps1` tried to make affordances for both `node` and `node.exe` existing, which is from https://github.com/npm/cli/pull/6548/files

`bin/npm` also makes the same check, so I opted for that here as well. I think it's more likely that the `cmd` scripts should be updated to do the same.

**Comment by lukekarrys** [comment]
We have tests for these files, but out of caution I am not going to land this until I'm able to at least simulate "installing" these files and running them on my Windows machine.

File: bin/npm.ps1 (hunk 2 above)
Comment (wraithgar) [reply bin/npm.ps1:5]: Good idea

**Review by wraithgar — APPROVED** [review]

**Pull request merged**